manifests/database.pp

```diff
@@ -24,12 +24,15 @@
     foreman::rake { 'db:migrate':
       unless => '/usr/sbin/foreman-rake db:abort_if_pending_migrations',
     }
-    ~> foreman_config_entry { 'db_pending_seed':
-      value => false,
-      dry   => true,
-    }
     ~> foreman::rake { 'db:seed':
       environment => delete_undef_values($seed_env),
+      unless      => '/usr/bin/test -f /etc/foreman/database_seeded',
```

Comment:
Puppet's exec has `creates` which natively implements this. Perhaps that should be exposed?

Reply:
I can go that route.  I was starting with not overloading the rake defines to see what you'd say before I went and added creates support :)

Reply:
I'm actually thinking about including the file creation in the rake task:

```puppet
define foreman::rake(
  Boolean $create_file = false,
) {
  exec { '...':
      # ...
  }

  if $create_file {
    file { "/var/lib/foreman/rake-${name}":
      ensure => file,
      # ...
      require => Exec['...'],
    }
  }
}

foreman::rake { 'db:seed':
  environment => delete_undef_values($seed_env),
  create_file => true,
}
```

On the other hand I'm also considering actually doing this in the seed job itself.

Reply:
Seed job itself? LIke a special rake task that runs seeds and creates the file? I considered that option as well.

Reply:
Yes, enhance the task. We already do so today: https://github.com/theforeman/foreman/blob/ae96cb70130968bdcc8b7a9a9af7edc7c6e0d9a8/lib/tasks/config.rake#L171-L184 (which needs to be replaced anyway if you're changing this in the installer)

Reply:
Opened a PR to core here -- https://github.com/theforeman/foreman/pull/7262